junit-runtime/src/main/scala/org/hamcrest/BaseDescription.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Ported from https://github.com/hamcrest/JavaHamcrest/
+ */
+package org.hamcrest
+
+import java.util.Arrays
+import java.lang.String.valueOf
+import scala.annotation.tailrec
+import org.hamcrest.internal.ArrayIterator
+import org.hamcrest.internal.SelfDescribingValueIterator
+
+abstract class BaseDescription extends Description {
+  override def appendText(text: String): Description = {
+    append(text)
+    this
+  }
+
+  override def appendDescriptionOf(value: SelfDescribing): Description = {
+    value.describeTo(this)
+    this
+  }
+
+  override def appendValue(value: AnyRef): Description = {
+    value match {
+      case null                   => append("null")
+      case v: String              => append(toJavaSyntax(v))
+      case v: java.lang.Character => append('"' + toJavaSyntax(v) + '"')
+      case v: java.lang.Short     => append(s"<${descriptionOf(value)}s>")
```

Comment:
It's not really possible to distinguish between `Byte`s, `Short`s and `Int`s at runtime in Scala.js. This case will harm the readability of most `Int`s. I would remove it.